lang/ruby/lib/avro/io.rb

```diff
@@ -397,7 +397,12 @@ def read_default_value(field_schema, default_value)
           return Integer(default_value)
         when :float, :double
           return Float(default_value)
-        when :boolean, :enum, :fixed, :string, :bytes
+        when :boolean, :enum, :fixed, :string
+          return default_value
+        when :bytes
+          if field_schema.logical_type == Schema::DECIMAL_LOGICAL_TYPE
+            return field_schema.type_adapter.decode(default_value)
```

Comment:
I wonder if we should apply `type_adapter.decode` to all values currently returned from this method (similar to the `read_data` method). There are logical types for `int` and `long` that would benefit from this fix too.

Reply:
07d18c67015f12218b0dc4d9f9c31e9e6271c48e